File: app/src/main/java/com/knopkapp/Registrationandverification2Fragment.java
package com.knopkapp;

import android.animation.ObjectAnimator;
import android.app.Activity;
import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.AccelerateDecelerateInterpolator;
import android.view.animation.DecelerateInterpolator;
import android.widget.Button;
import android.widget.ProgressBar;

import com.knopkapp.activities.WriteSMSFragment;

public class Registrationandverification2Fragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_registrationandverification2, container, false);
        updateProgressBar(60);

        Button nextButton = view.findViewById(R.id.regnext2);
        nextButton.setOnClickListener(v -> replaceFragment(new Registrationandverification3Fragment()));

        return view;
    }

    private void replaceFragment(Fragment fragment) {
        getActivity().getSupportFragmentManager().beginTransaction()
                .replace(R.id.fragmentContainer, fragment)
                .addToBackStack(null)
                .commit();
    }

    private void updateProgressBar(int progress) {
        Activity activity = getActivity();
        if (activity != null) {
            ProgressBar progressBar = activity.findViewById(R.id.progressBar);

            ObjectAnimator animation = ObjectAnimator.ofInt(progressBar, "progress", progressBar.getProgress(), progress);
            animation.setDuration(1000);
            animation.setInterpolator(new AccelerateDecelerateInterpolator());
            animation.start();
        }
    }
}

File: app/src/main/java/com/knopkapp/Registrationandverification2Fragment.kt
package com.knopkapp

import android.animation.ObjectAnimator
import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.AccelerateDecelerateInterpolator
import android.widget.Button
import android.widget.ProgressBar
import androidx.fragment.app.Fragment
import androidx.navigation.fragment.findNavController
import com.knopkapp.databinding.FragmentRegistrationandverification2Binding

class Registrationandverification2Fragment : Fragment() {
    private lateinit var binding: FragmentRegistrationandverification2Binding
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = FragmentRegistrationandverification2Binding.inflate(layoutInflater)
        updateProgressBar(60)
        binding.regnext2.setOnClickListener {

            findNavController().navigate(R.id.registrationandverification3Fragment)

        }
        return binding.root
    }

    /*  private fun replaceFragment(fragment: Fragment) {
          activity!!.supportFragmentManager.beginTransaction()
              .replace(R.id.fragmentContainer, fragment)
              .addToBackStack(null)
              .commit()
      }*/

    private fun updateProgressBar(progress: Int) {
        val activity: Activity? = activity
        if (activity != null) {
            val progressBar = activity.findViewById<ProgressBar>(R.id.progressBar)
            val animation =
                ObjectAnimator.ofInt(progressBar, "progress", progressBar.progress, progress)
            animation.duration = 1000
            animation.interpolator = AccelerateDecelerateInterpolator()
            animation.start()
        }
    }
}